desktop/core/src/desktop/lib/fs/ozone/upload.py

```diff
@@ -29,9 +30,110 @@
 else:
   from django.utils.translation import ugettext as _
 
+from desktop.lib.exceptions_renderable import PopupException
+from filebrowser.utils import generate_chunks
 
 LOG = logging.getLogger()
 
+class OFSFineUploaderChunkedUpload(object):
+  def __init__(self, request, *args, **kwargs):
+    self.qquuid = kwargs.get('qquuid')
+    self.qqtotalparts = kwargs.get('qqtotalparts')
+    self.totalfilesize = kwargs.get('qqtotalfilesize')
+    self.file_name = kwargs.get('qqfilename')
+    self.chunk_size = UPLOAD_CHUNK_SIZE.get()
+    self.destination = kwargs.get('dest', None)  # GET param avoids infinite looping
+    self.target_path = None
+    self.file = None
+    self._request = request
+    self._part_size = UPLOAD_CHUNK_SIZE.get()
+
+  def check_access(self):
+    if self._is_ofs_upload():
+      self._fs = self._get_ofs(self.request)
+
+      # Verify that the path exists
+      try:
+        self._fs.stats(self.destination)
+      except Exception as e:
+        raise PopupException(_('Destination path does not exist: %s' % self.destination))
+
+      LOG.debug("Chunk size = %d" % UPLOAD_CHUNK_SIZE.get())
+      LOG.info('Using OFSFileUploadHandler to handle file upload.')
+      self.target_path = self._fs.join(self.destination, self.file_name)
+
+  def upload_chunks(self):
+    LOG.debug("OFSfileUploadHandler receive_data_chunk")
+    try:
+      LOG.debug("OFSFileUploadHandler uploading file part with size: %s" % self._part_size)
+      fp = io.BytesIO()
+      for i, (chunk, total) in enumerate(generate_chunks(self.qquuid, self.qqtotalparts, self.chunk_size),1):
+        logging.debug("OFSfileUploadHandler uploading file %s, part %d, size %d, dest: %s" % (self.file_name, i, total, self.dest))
+        fp.write(chunk)
```

Comment:
Does this end up writing all the bytes to fp (in-memory)? If that's true that it could be problematic with large files.